habit/models.py

```diff
@@ -0,0 +1,46 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+from datetime import datetime
+from django.db.models import UniqueConstraint
+from django.db.models import Q
+
+
+class User(AbstractUser):
+    def __repr__(self):
+        return f"<User username={self.username} pk={self.pk}>"
+    def __str__(self):
+        return self.username
+
+class HabitManager(models.Manager):
+    def for_user(self, user):
+        if user.is_authenticated:
+            habits=self.filter(Q( public=True) | Q(author=user))
+        else:
+            habits=self.filter(public=True)
+        return habits
```

Comment:
You don't need this -- the fields `public` and `author` are specific to the recipe model from the example.